In [1]:
import pandas as pd
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

## Set up data

In [2]:
columns = ['Database Name', 'Patient Number', 'Client Number', 'Consultation Number', 'Item Name', 'Units',
           'Item Label', 'Clinic Code', 'Consulting Vet', 'Patient Desexed', 'Deceased Date',
           'Visit Date', 'Consultation Date', 'Examination Text' ,'Patient Species', 'Patient Breed',
           'Patient Colour', 'Patient Sex', 'Patient Date of Birth', 'Insured', 'weight', 'temperature',
           'HeartRate', 'BodyScore', 'DentalGrade', 'RespRate', 'PainScore', 'BP', 'CRT', 'MMColour']

In [3]:
data_dir = '../Data/vet_compass/'
data = pd.read_csv(data_dir + 'STR024A03 20190719 sample x10,000.csv', names = columns)

/anaconda3/envs/vetAI/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3057: DtypeWarning: Columns (1,2,3,19,20,21,22,25,27) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


## Explore vital signs columns

In [4]:
vitalsigns = ['weight', 'temperature', 'HeartRate', 'BodyScore', 'DentalGrade', 'RespRate', 'PainScore', 'BP', 'CRT', 'MMColour']

for sign in vitalsigns:
    print(data[sign].dtype)

object
object
object
object
object
object
object
object
object
object


In [5]:
data[vitalsigns].describe()

weight temperature HeartRate BodyScore DentalGrade RespRate PainScore  \
count   318629      112138    117336     99396       94292    70632     22699   
unique    1920         340       406       187         308      283       113   
top        4.5        38.6       180       5/9           0       24         0   
freq      5908        8547     20327     20524       27480     9030     13815   

          BP     CRT MMColour  
count   3089  102755   112096  
unique   123     202      358  
top        -      <2     pink  
freq    1566   31601    70763

In [6]:
data_count = data[vitalsigns].count().reset_index(name='count').sort_values(['count'], ascending=False)
data_count['proportion'] = data_count['count']/data.shape[0]

data_count

index   count  proportion
0       weight  318629    0.440918
2    HeartRate  117336    0.162369
1  temperature  112138    0.155176
9     MMColour  112096    0.155118
8          CRT  102755    0.142192
3    BodyScore   99396    0.137544
4  DentalGrade   94292    0.130481
5     RespRate   70632    0.097740
6    PainScore   22699    0.031411
7           BP    3089    0.004275

Weight is the vital sign with the largest number of non-nan entries (44%), followed by heart rate (16%). Unfortunately, heart rate and blood pressure could be strongly influenced by being at the vet, so these may not be so useful. Worth looking into heart rate anyway, given data availability.